test/test_auth_api.js

```diff
@@ -0,0 +1,228 @@
+import btoa       from 'btoa';
+import jwt        from 'jsonwebtoken';
+import nock       from 'nock';
+import should     from 'should';
+import supertest  from 'supertest-as-promised';
+import url        from 'url';
+
+import app        from '../src/server';
+import db         from '../src/models/db';
+import config     from '../src/config';
+import {
+  errnos,
+  ERRNO_UNAUTHORIZED,
+  errors,
+  UNAUTHORIZED
+} from '../src/errors';
+
+import {
+  loginAsAdmin,
+  loginAsClient,
+  createClient,
+} from './common';
+
+const endpointPrefix = '/' + config.get('version');
+const server = supertest(app);
```

Comment:
it's not an agent anymore; an agent keeps state between calls and I think we should do it only when needed.